pkg/scheduler/service/service.go

```diff
@@ -492,17 +484,36 @@ func (s *SchedulerServer) awaitFailover(schedulerProcess schedulerProcessInterfa
 	return <-errCh
 }
 
-func validateLeadershipTransition(desired, current string) {
+// validateLeadershipTransition logs and panics if the leadership transition fails.
+// Ideally an external process monitor will re-start the process.
+func validateLeadershipTransition(desired, current string) error {
 	log.Infof("validating leadership transition")
-	d := uid.Parse(desired).Group()
-	c := uid.Parse(current).Group()
-	if d == 0 {
+	desiredUID, err := uid.Parse(desired)
+	if err != nil {
+		return fmt.Errorf("Failed to parse desired scheduler UID %q: %v", desired, err)
+	}
+
+	desiredGroup := desiredUID.Group()
+	if desiredGroup == 0 {
```

Comment:
since you're already erroring out on parse errors, why another check for 0 here?